tests/TestCase/Command/PluginAssetsCommandsTest.php

```diff
@@ -76,7 +76,7 @@ public function tearDown(): void
      */
     public function testSymlink(): void
     {
-        $this->loadPlugins(['TestPlugin' => ['routes' => false], 'Company/TestPluginThree']);
+        Configure::write('Plugins.autoload', ['TestPlugin' => ['routes' => false], 'Company/TestPluginThree' => []]);
```

Comment:
This is a breaking change to applications/plugins too as what used to work (calling `loadPlugins()` no longer does). Perhaps we should revert the exception on double loading plugins and take a different approach?

Reply:
Maybe I misunderstand the loadPlugins method but all it does is create a fake app object and load the specified plugins (which therefore adjusts the static PluginCollection)

All this does is offload the plugin loading from the fake application object to the "real" one being used in the integration test.

If loadPlugins was broken, there would be many many other tests failing as well. its just in those specific cases where this adjustment was needed.

Reply:
I checked like 28 of my plugins, and none of them use loadPlugins()
Are we talking about test harness or also src/ code problems for plugins?
If this is only about tests, it seems a breaking change is not that big of a deal as long as the change is documented in the guide and the benefit is clear - which in this case seems to be.

Reply:
loadPlugins() is primarily used in the core tests only since we needs to tests various stuff with different sets of plugins loaded. There are still numerous tests in core using it besides the one LordSimal modified so it works just fine. Apps would rarely need to use this method as all the plugins it needs would be loaded in application bootstrap itself.

Reply:
this `loadPlugins()` method is only used in tests